shared/middleware/helpers/bucket_helper.rb

```diff
@@ -273,16 +273,21 @@ def restore_previous_version(encrypted_channel_id, filename, version_id, user_id
 
     version_restored = false
 
-    begin
-      response = s3.copy_object(
-        bucket: @bucket,
-        key: key,
-        copy_source: "#{@bucket}/#{key}?versionId=#{version_id}"
-      )
-      version_restored = true
-    rescue Aws::S3::Errors::NoSuchVersion, Aws::S3::Errors::InvalidArgument => err
-      raise err unless err.is_a?(Aws::S3::Errors::NoSuchVersion) || invalid_version_id?(err)
+    unless version_id.nil? || version_id.empty?
+      begin
+        response = s3.copy_object(
+          bucket: @bucket,
+          key: key,
+          copy_source: "#{@bucket}/#{key}?versionId=#{version_id}"
+        )
+        version_restored = true
+      rescue Aws::S3::Errors::NoSuchVersion, Aws::S3::Errors::InvalidArgument => err
+        raise err unless err.is_a?(Aws::S3::Errors::NoSuchVersion) || invalid_version_id?(err)
```

Comment:
This might read better as:
```
  begin
     ...
  rescue Aws::S3::Errors::NoSuchVersion
    # do nothing
  rescue Aws::S3::Errors::InvalidArgument => err
    raise unless invalid_version_id?(err)
  end
```